### PR TITLE
5 new endpoints: PATCH appClient, batch, model, GET model, and POST models

### PR DESCRIPTION
**Pull request by magnurud**

### CODE REVIEW

File: Lucidtech/Las/Client.cs [hunk 1]
```diff
@@ -610,6 +650,52 @@ public object CreateBatch(string? name = null, string? description = null)
             return ExecuteRequestResilient(RestSharpClient, request);
         }
 
+        /// <summary>Creates a model, calls the POST /models endpoint.</summary>
+        /// <param name="width">The number of pixels to be used for the input image width of your model</param>
+        /// <param name="height">The number of pixels to be used for the input image height of your model</param>
+        /// <param name="fieldConfig">Specification of the fields that the model is going to predict</param>
+        /// <param name="preprocessConfig">Specification of the processing steps prior to the prediction of an image</param>
+        /// <param name="name">Name of the model</param>
+        /// <param name="description>Description of the model</param>
```
Comment (tringholm) [inline Lucidtech/Las/Client.cs:659]: Missing quote

File: Lucidtech/Las/Client.cs [hunk 2]
```diff
@@ -638,6 +724,76 @@ public object CreateBatch(string? name = null, string? description = null)
...
+        }
+
+        /// <summary>Updates a model, calls the PATCH /models/{modelId} endpoint.</summary>
+        /// <param name="modelId">Id of the model</param>
+        /// <param name="width">The number of pixels to be used for the input image width of your model</param>
+        /// <param name="height">The number of pixels to be used for the input image height of your model</param>
+        /// <param name="fieldConfig">Specification of the fields that the model is going to predict</param>
+        /// <param name="preprocessConfig">Specification of the processing steps prior to the prediction of an image</param>
+        /// <param name="name">Name of the model</param>
+        /// <param name="description>Description of the model</param>
```
Comment (tringholm) [inline Lucidtech/Las/Client.cs:743]: Missing quote

File: Lucidtech/Lucidtech.csproj [hunk 3]
```diff
@@ -15,7 +15,7 @@
     <TargetFrameworks>netstandard20</TargetFrameworks>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
     <PackageId>Lucidtech.Las</PackageId>
-    <PackageVersion>1.2.0</PackageVersion>
+    <PackageVersion>1.2.1</PackageVersion>
```
Comment (tringholm) [inline Lucidtech/Lucidtech.csproj:18]: Should the version be 1.3.0? This update includes new non-breaking features, so I'd expect a jump in minor version

File: Lucidtech/Las/Client.cs [hunk 4]
```diff
@@ -53,23 +53,23 @@ public Client(Credentials credentials)
         /// <param name="attributes">Additional attributes</param>
         /// <returns>AppClient response from REST API</returns>
         public object CreateAppClient(
-            bool generateSecret = true, 
-            List<string>? logoutUrls = null, 
+            bool generateSecret = true,
+            List<string>? logoutUrls = null,
             List<string>? callbackUrls = null,
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:58]: Missing loginUrls, default_login_url

File: Lucidtech/Las/Client.cs (hunk 1 above)
Comment (augustak) [inline Lucidtech/Las/Client.cs:659]:
```suggestion
        /// <param name="description">Description of the model</param>
```

File: Lucidtech/Las/Client.cs (hunk 2 above)
Comment (augustak) [inline Lucidtech/Las/Client.cs:743]:
```suggestion
        /// <param name="description">Description of the model</param>
```

File: Lucidtech/Las/Client.cs [hunk 5]
```diff
@@ -638,6 +724,76 @@ public object CreateBatch(string? name = null, string? description = null)
             RestRequest request = ClientRestRequest(Method.GET, "/models", null, queryParams);
             return ExecuteRequestResilient(RestSharpClient, request);
         }
+        
+        /// <summary>Get information about a specific model, calls the GET /models/{model_id} endpoint.</summary>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:728]:
```suggestion
        /// <summary>Get information about a specific model, calls the GET /models/{modelId} endpoint.</summary>
```

File: Lucidtech/Las/Client.cs [hunk 6]
```diff
@@ -638,6 +724,76 @@ public object CreateBatch(string? name = null, string? description = null)
...
+
+        /// <summary>Updates a model, calls the PATCH /models/{modelId} endpoint.</summary>
+        /// <param name="modelId">Id of the model</param>
+        /// <param name="width">The number of pixels to be used for the input image width of your model</param>
+        /// <param name="height">The number of pixels to be used for the input image height of your model</param>
+        /// <param name="fieldConfig">Specification of the fields that the model is going to predict</param>
+        /// <param name="preprocessConfig">Specification of the processing steps prior to the prediction of an image</param>
+        /// <param name="name">Name of the model</param>
+        /// <param name="description>Description of the model</param>
+        /// <param name="attributes">Additional attributes</param>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:744]: Missing status param

File: Lucidtech/Lucidtech.csproj (hunk 3 above)
Comment (augustak) [inline Lucidtech/Lucidtech.csproj:18]:
```suggestion
    <PackageVersion>1.3.0</PackageVersion>
```

**Review by tringholm — CHANGES_REQUESTED** [review]

File: Test/Service.cs [hunk 7]
```diff
@@ -96,6 +98,32 @@ public static class Util {
...
+                    {"description", "due date of the invoice"},
+                    {"maxLength", 11}
+                    }
+                },
+                {"TotalAmount", new Dictionary<string, object>(){
+                    {"type", "amount"},
+                    {"description", "total amount of the invoice"},
+                    {"maxLength", 15}
+                    }
+                },
```
Comment (augustak) [inline Test/Service.cs:123]: Indentation here looks a bit off

File: Test/Service.cs [hunk 8]
```diff
@@ -96,6 +98,32 @@ public static class Util {
...
+            return new Dictionary<string, object>{
+                {"autoRotate", true},
+                {"maxPages", 3},
+                {"imageQuality", "HIGH"}
+            };
+        }
+            
+        public static Dictionary<string, object> FieldConfig(){
+            return new Dictionary<string, object>{
+                {"DueDate", new Dictionary<string, object>(){
```
Comment (augustak) [inline Test/Service.cs:112]: Do we need parenthesises?

File: Test/Service.cs [hunk 9]
```diff
@@ -96,6 +98,32 @@ public static class Util {
...
+                    {"type", "date"},
+                    {"description", "due date of the invoice"},
+                    {"maxLength", 11}
+                    }
+                },
+                {"TotalAmount", new Dictionary<string, object>(){
```
Comment (augustak) [inline Test/Service.cs:118]: (Foreach Dictionary in code) Do we not want a nice litte whitespace between > and {? Like so:

```suggestion
                {"TotalAmount", new Dictionary<string, object> {
```

**Review by augustak — CHANGES_REQUESTED** [review]

File: Lucidtech/Las/Client.cs [hunk 10]
```diff
@@ -34,42 +34,43 @@ public Client(Credentials credentials)
...
-        ///     attributes: parameters, 
-        ///     generateSecret: false,
-        ///     logoutUrls: new List&lt;string&gt;{"https://localhost:3030/logout"},
-        ///     callbackUrls: new List&lt;string&gt;{"https://localhost:3030/callback"}
-        /// );
-        /// </code>
-        /// </example>
+        /// <param name="generateSecret">Set to false to ceate a Public app client, default: true</param>
+        /// <param name="logoutUrls">List of logout urls</param>
+        /// <param name="callbackUrls>List of callback urls</param>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:41]:
```suggestion
        /// <param name="callbackUrls">List of callback urls</param>
```

File: Test/Service.cs [hunk 11]
```diff
@@ -71,11 +73,11 @@ public static class Util {
         }
         
         public static Dictionary<string, object> CompletedConfig(){
```
Comment (augustak) [inline Test/Service.cs:75]:
```suggestion
        public static Dictionary<string, object> CompletedConfig() {
```

File: Test/Service.cs [hunk 12]
```diff
@@ -84,18 +86,45 @@ public static class Util {
         }
             
         public static Dictionary<string, object> ErrorConfig(){
```
Comment (augustak) [inline Test/Service.cs:88]:
```suggestion
        public static Dictionary<string, object> ErrorConfig() {
```

File: Test/Service.cs [hunk 13]
```diff
@@ -84,18 +86,45 @@ public static class Util {
...
-            return new Dictionary<string, object>{
+            return new Dictionary<string, object> {
                 {"email", "foo@lucidtech.io"},
                 {"manualRetry", true}
             };
         }
             
         public static Dictionary<string, string?> NameAndDescription(string? name, string? description){
```
Comment (augustak) [inline Test/Service.cs:95]:
```suggestion
        public static Dictionary<string, string?> NameAndDescription(string? name, string? description) {
```

File: Test/Service.cs [hunk 14]
```diff
@@ -84,18 +86,45 @@ public static class Util {
...
-            return new Dictionary<string, string?>{
+            return new Dictionary<string, string?> {
                 {"name", name},
                 {"description", description}
             };
         }
+        
+        public static Dictionary<string, object> PreprocessConfig(){
```
Comment (augustak) [inline Test/Service.cs:102]:
```suggestion
        public static Dictionary<string, object> PreprocessConfig() {
```

File: Test/Service.cs [hunk 15]
```diff
@@ -84,18 +86,45 @@ public static class Util {
...
+            return new Dictionary<string, object> {
+                {"autoRotate", true},
+                {"maxPages", 3},
+                {"imageQuality", "HIGH"}
+            };
+        }
+            
+        public static Dictionary<string, object> FieldConfig(){
```
Comment (augustak) [inline Test/Service.cs:110]:
```suggestion
        public static Dictionary<string, object> FieldConfig() {
```

**Review by tringholm — APPROVED** [review]

**Review by augustak — APPROVED** [review]

**Pull request merged**